app/controllers/comments_controller.rb

```diff
@@ -12,8 +15,37 @@ def create
             render :new, status: :unprocessable_entity
         end
     end
+
+    # Sends a CSV file containing an export of all visible comments of visible posts, ordered 
+    # ascendingly by the time of the comments' creation, with three columns: 
+    # "post title", "comment author", "comment body"
+    def export
+        current_datetime_string = DateTime.now.utc.strftime("%Y-%d-%m_%H-%M-%S")
+
+        send_data generate_export_csv,
+            filename: "all_comments_export_#{current_datetime_string}.csv",
+            type: "text/csv"
+    end
+
     private
-        def comment_params
-            params.require(:comment).permit(:author, :body, :status)
+
+    def comment_params
+        params.require(:comment).permit(:author, :body, :status)
+    end
+
+    def generate_export_csv
+        header_row = ["post title", "comment author", "comment body"]
+
+        data_rows = Comment
+            .joins(:post)
+            .where.not('comments.status': 'archived')
+            .where.not('posts.status': 'archived')
+            .order('comments.created_at': :asc)
+            .pluck(:'posts.title', :'comments.author', :'comments.body')
```

Comment:
Just as a point of discussion: First I had something like this here, which reads a little nicer, I think, but is less efficient, since it fetches a lot of data from the DB that's not needed.
```ruby
        data_rows = Comment
            .order(created_at: :asc)
            .preload(:post)
            .reject {|c| c.archived? or c.post.archived?}
            .map {|c| [c.post.title, c.author, c.body]}
```